src/ring_theory/graded_algebra/basic.lean

```diff
@@ -39,6 +44,17 @@ variables {ι R A : Type*}
 variables [decidable_eq ι] [add_comm_monoid ι] [comm_semiring R] [ring A] [algebra R A]
 variables (𝒜 : ι → submodule R A)
 
+lemma direct_sum.coe_mul_apply_submodule [set_like.graded_monoid 𝒜]
+  [Π (i : ι) (x : (λ (i : ι), ↥(𝒜 i)) i), decidable (x ≠ 0)] (r r' : ⨁ i, 𝒜 i) (i : ι) :
+  ((r * r') i : A) =
+    ∑ ij in finset.filter (λ ij : ι × ι, ij.1 + ij.2 = i) (r.support.product r'.support),
+      r ij.1 * r' ij.2 :=
+begin
+  rw [direct_sum.mul_eq_sum_support_ghas_mul, dfinsupp.finset_sum_apply, submodule.coe_sum],
+  simp_rw [direct_sum.coe_of_submodule_apply, ←finset.sum_filter, set_like.coe_ghas_mul],
+end
```

Comment:
This one probably belongs in `direct_sum/algebra`, since it's about `graded_monoid` not about `graded_algebra`

Reply:
And maybe it's worth copying this for the `add_subgroup` and `add_submonoid` versions too, since the changes should be trivial to make.